create_venv.py

```diff
@@ -0,0 +1,127 @@
+venv_activate_this = os.path.join(venv_script_path, 'activate_this.py')
+venv_python_bin = os.path.join(venv_script_path, 'python')
+
+usage_message = """
+***********************************************************************
+To run the Firefox UI Tests, activate the virtual environment:
+    {}{}
+
+See firefox-ui-tests --help for all options
+
+***********************************************************************
+"""
+
+
+def download(url, target):
+    """Downloads the specified url to the given target."""
+    response = urllib2.urlopen(url)
+    with open(target, 'wb') as f:
+        f.write(response.read())
+
+    return target
+
+
+def create_virtualenv(target, python_bin=None):
+    script_path = os.path.join(here, 'virtualenv-%s' % VIRTUALENV_VERSION,
+                               'virtualenv.py')
+
+    print 'Downloading virtualenv %s' % VIRTUALENV_VERSION
+    zip_path = download(VIRTUALENV_URL % {'VERSION': VIRTUALENV_VERSION},
+                        os.path.join(here, 'virtualenv.zip'))
+
+    try:
+        with zipfile.ZipFile(zip_path, 'r') as f:
+            f.extractall(here)
+
+        print 'Creating new virtual environment'
+        cmd_args = [sys.executable, script_path, target]
+
+        if python_bin:
+            cmd_args.extend(['-p', python_bin])
+
+        subprocess.check_call(cmd_args)
+    finally:
+        try:
+            os.remove(zip_path)
+        except OSError:
+            pass
+
+        shutil.rmtree(os.path.dirname(script_path), ignore_errors=True)
+
+
+def main():
+    parser = optparse.OptionParser('Usage: %prog [options] path_to_venv')
+    parser.add_option('-p', '--python',
+                      type='string',
+                      dest='python',
+                      metavar='BINARY',
+                      help='The Python interpreter to use.')
+    parser.add_option('--with-requirements',
+                      dest='with_requirements',
+                      action='store_true',
+                      help='Install optional packages as listed in requirements.txt')
+
+    (options, args) = parser.parse_args(args=None, values=None)
+
+    if len(args) != 1:
+        parser.error('Path to the environment has to be specified')
+    target = args[0]
+    assert(target)
+
+    # Remove an already existent virtual environment
+    if os.path.exists(target):
+        print 'Removing already existent virtual environment at: %s' % target
+        shutil.rmtree(target, True)
+
+    create_virtualenv(target, python_bin=options.python)
+
+    # Activate the environment
+    tps_env = os.path.join(target, venv_activate_this)
+    execfile(tps_env, dict(__file__=tps_env))
+
+    # Install Firefox UI Tests package
+    subprocess.check_call([os.path.join(target, venv_python_bin),
+                           os.path.join(here, 'setup.py'), 'develop'])
+
+    # Install optional packages via requirements.txt
+    if options.with_requirements:
+        subprocess.check_call(['pip', 'install', '-r',
```

Comment:
No, you don't have. At this point we are inside the virtualenv and setuptools and pip will be available.